File: paypay/javascript/qr.js
'use strict';
require('dotenv').config();
const PAYPAY = require('@paypayopa/paypayopa-sdk-node');
PAYPAY.Configure({
  clientId: process.env.CLIENT_ID,
  clientSecret: process.env.CLIENT_SEACRET,
  merchantId: process.env.MERCHANT_ID,
  productionMode: false,
});
var rand = Math.floor( Math.random() * 100 ) ;
let payload = {
    merchantPaymentId: new Date().toISOString(),
    amount: {
      amount: 100,
      currency: "JPY"
    },
    orderDescription: '説明',
    codeType: "ORDER_QR",
    orderDescription: "メニュー名",
    isAuthorization: false,
    redirectUrl: "https://miniapps.paypay.ne.jp/",
    redirectType: "WEB_LINK",
  };


const open = require('open');
PAYPAY.QRCodeCreate(payload, (response) => {
  let body_arr = JSON.parse(response.BODY);
  if(body_arr.resultInfo.code === 'SUCCESS'){
    open(body_arr.data.url);
  }
});

File: paypay/python/qr.py
import paypayopa
import time

API_KEY = 'a_K86OGgHiIW_kX6a'
API_SECRET = 'BErO7ESgBqWz7nSeSO+FqZIp/pUMXdeJfY2E2pUG'
client = paypayopa.Client(auth=(API_KEY, API_SECRET), production_mode=False)
client.set_assume_merchant("302793184549576704")

request = {
    "merchantPaymentId": round(time.time()),
    "codeType": "ORDER_QR",
    "redirectUrl": "http://foobar.com",
    "redirectType": "WEB_LINK",
    "orderDescription":"Example - Mune Cake shop",
    "orderItems": [{
        "name": "Moon cake",
        "category": "pasteries",
        "quantity": 1,
        "productId": "67678",
        "unitPrice": {
            "amount": 1,
            "currency": "JPY"
        }
    }],
    "amount": {
        "amount": 1,
        "currency": "JPY"
    },
}

response = client.Code.create_qr_code(request)
print(response['data']['url'])